src/pybind/rados.py

```diff
@@ -1821,16 +1847,204 @@ def lookup_snap(self, snap_name):
 
     def get_last_version(self):
         """
-        Return the version of the last object read or written to.
+            Return the version of the last object read or written to.
 
-        This exposes the internal version number of the last object read or
-        written via this io context
+            This exposes the internal version number of the last object read or
+            written via this io context
 
-        :returns: version of the last object used
+            :returns: version of the last object used
         """
         self.require_ioctx_open()
         return run_in_thread(self.librados.rados_get_last_version, (self.io,))
 
+    def create_write_op(self):
+        """
+            create write operation object.
+            need call release_write_op after use
+        """
+        self.librados.rados_create_write_op.restype = c_void_p
+        return run_in_thread(self.librados.rados_create_write_op, (None,))
+
+    def create_read_op(self):
+        """
+           create read operation object.
+           need call release_read_op after use
+        """
+        self.librados.rados_create_read_op.restype = c_void_p
+        return run_in_thread(self.librados.rados_create_read_op, (None,))
+
+
+    def release_write_op(self, write_op):
+        """
+           release memory alloc by create_write_op
+        """
+        self.librados.rados_release_write_op.argtypes = [c_void_p]
+        run_in_thread(self.librados.rados_release_write_op, (c_void_p(write_op),))
+
+    def release_read_op(self, read_op):
+        """
+           release memory alloc by create_read_op
+           :para read_op: read_op object
+           :type: int
+        """
+        self.librados.rados_release_read_op.argtypes = [c_void_p]
+        run_in_thread(self.librados.rados_release_read_op, (c_void_p(read_op),))
+
+    @requires(('write_op', int), ('keys', tuple), ('values', tuple))
+    def set_omap(self, write_op, keys, values):
+        """
+           set keys values to write_op
+           :para write_op :write_operation object
+           :type int
+           :para keys :a tuple of keys
+           :type tuple
+           :para values :a tuple of values
+           :type tuple
+           :para value_lens :lens of values
+           :type tuple
+           :para count: total counts of key-value pairs
+        """
+        if len(keys) != len(values):
+            raise Error("Rados(): keys and values must hava the same numbers")
```

Comment:
i want make the input more simple ,so i have count these two para in function,but forget to delete the comment